File: PHP/122. Best Time to Buy and Sell Stock II.php
<?php

/**
 * @noinspection PhpIllegalPsrClassPathInspection
 * @noinspection PhpMultipleClassDeclarationsInspection
 * @noinspection PhpUnused
 */
class Solution
{
    /**
     * @param  int[]  $prices
     */
    public function maxProfit(array $prices): int
    {
        $profit = 0;

        foreach ($prices as $price) {
            if (isset($prev)) {
                $profit += max(0, $price - $prev);
            }

            $prev = $price;
        }

        return $profit;
    }
}

File: Go/122. Best Time to Buy and Sell Stock II.go
package main

//goland:noinspection GoUnusedFunction
func maxProfitTwo(prices []int) int {
	profit := 0
	prev := prices[0]

	for _, price := range prices {
		if price > prev {
			profit += max(0, price-prev)
		}

		prev = price
	}

	return profit
}
